numba/cuda/tests/cudadrv/test_linker.py

```diff
@@ -95,6 +97,76 @@ def test_linking_lazy_compile(self):
     def test_linking_eager_compile(self):
         self._test_linking(eager=True)
 
+    @skip_unless_cuda_python('NVIDIA Binding needed for NVRTC')
+    def test_linking_cu(self):
+        bar = cuda.declare_device('bar', 'int32(int32)')
+
+        link = os.path.join(os.path.dirname(__file__), 'data', 'jitlink.cu')
+
+        @cuda.jit(link=[link])
+        def kernel(r, x):
+            i = cuda.grid(1)
+
+            if i < len(r):
+                r[i] = bar(x[i])
+
+        x = np.arange(10, dtype=np.int32)
+        r = np.zeros_like(x)
+
+        kernel[1, 32](r, x)
+
+        # Matches the operation of bar() in jitlink.cu
+        expected = x * 2
+        np.testing.assert_array_equal(r, expected)
+
+    @skip_unless_cuda_python('NVIDIA Binding needed for NVRTC')
+    def test_linking_cu_log_warning(self):
+        bar = cuda.declare_device('bar', 'int32(int32)')
+
+        link = os.path.join(os.path.dirname(__file__), 'data', 'warn.cu')
+
+        with warnings.catch_warnings(record=True) as w:
+            @cuda.jit('void(int32)', link=[link])
```

Comment:
context managed block should probably have a call to this to filter known internal warnings:
https://github.com/numba/numba/blob/59fa2e335be68148b3bd72a29de3ff011430038d/numba/tests/support.py#L141-L144

Reply:
I've now added the call.